plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -624,7 +624,7 @@ func PodFitsResources(pod *v1.Pod, meta interface{}, nodeInfo *schedulercache.No
 // terms are ORed, and an empty list of terms will match nothing.
 func nodeMatchesNodeSelectorTerms(node *v1.Node, nodeSelectorTerms []v1.NodeSelectorTerm) bool {
 	for _, req := range nodeSelectorTerms {
-		nodeSelector, err := v1helper.NodeSelectorRequirementsAsSelector(req.MatchExpressions)
```

Comment:
You need to validate it at least once, but it can be done on a higher level. Why not passed a selector here instead of nodeSelectorTerms?